src/main/java/com/google/step/datamanager/DealTagManagerDatastore.java

```diff
@@ -87,4 +90,14 @@ private Entity createDealTagEntity(long dealId, long tagId) {
     entity.setProperty("tagId", tagId);
     return entity;
   }
+
+  @Override
+  public void deleteAllTagsOfDeal(long dealId) {
+    Filter propertyFilter = new FilterPredicate("dealId", FilterOperator.EQUAL, dealId);
+    Query query = new Query("DealTag").setFilter(propertyFilter).setKeysOnly();
+    PreparedQuery pq = datastore.prepare(query);
+    List<Key> keys = new ArrayList<>();
+    pq.asIterable().forEach(entity -> keys.add(entity.getKey()));
+    datastore.delete(keys);
```

Comment:
```suggestion
    pq.asIterable().forEach(entity -> datastore.delete(entity.getKey()));
```
would this be better?

Reply:
no, it's better to call delete(many things) instead of delete(one key) many times

Reply:
It might be better to use pq.asList.map(), because I am not sure whether the list.add operation in stream can lead to race condition. I'll check are there any other places are using this.